src/libraries/System.Private.Xml/src/System/Xml/XPath/Internal/FollowingQuery.cs

```diff
@@ -47,7 +49,7 @@ public override XPathNavigator Advance()
             while (!_iterator.MoveNext())
             {
                 bool matchSelf;
-                if (_input.NodeType == XPathNodeType.Attribute || _input.NodeType == XPathNodeType.Namespace)
+                if (_input!.NodeType == XPathNodeType.Attribute || _input.NodeType == XPathNodeType.Namespace)
```

Comment:
same, I'd expect this not to be needed

Reply:
I don't follow on this one, I cannot remove it even after applying above suggestion.

Reply:
I meant that you have a null check on top which means there is a possibility of this somehow becoming null and throwing NRE or something likely can be annotated in more expressive way (assuming such annotation exists). We need to figure out why compiler thinks it can be null and fix annotations related to that line of code

Reply:
(I don't feel strongly on fixing this since it won't affect public APIs but such cases may suggest product bug)

Reply:
> I meant that you have a null check on top which means there is a possibility of this somehow becoming null

Which null check are you referring to? The compiler complains probably because in the case that `_iterator` is not null beforehand, we don't perform a null check of `_input`. 

I added the bang (`!`) based on the assumption that `_input != null` when `_iterator != null`.